### PR TITLE
Use any config for plugins

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Use ov::Any for configs
 - Needed for PR https://github.com/openvinotoolkit/openvino/pull/10986 because I am caching the original config

### Tickets:
 - 78226

### CODE REVIEW

File: src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp [hunk 1]
```diff
@@ -207,6 +207,10 @@ void IInferencePlugin::SetConfig(const std::map<std::string, std::string>&) {
     IE_THROW(NotImplemented);
 }
 
+void IInferencePlugin::SetProperties(const ov::AnyMap&) {
+    IE_THROW(NotImplemented);
```
[the remark this answers is not in the document]
Comment (ilyachur) [reply src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp:211]: Good catch! Will change.

[the remark this answers is not in the document]
Comment (ilyachur) [reply src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp]: Fixed

File: src/inference/src/ie_core.cpp [hunk 2]
```diff
@@ -617,7 +617,11 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
                 IE_THROW() << "Device name must not contain dot '.' symbol";
             }
             const auto& value = plugin.second;
-            PluginDescriptor desc{value.m_create_plugin_func, value.m_default_config, value.m_create_extension_func};
+            ov::AnyMap config;
+            for (const auto& it : value.m_default_config) {
+                config[it.first] = it.second;
+            }
```
Comment (ilya-lavrenov) [inline src/inference/src/ie_core.cpp:623]: can we create `any_copy` shortcut here? To hide boilerplate code

Comment (ilyachur) [reply src/inference/src/ie_core.cpp:623]: any_copy creates `std::map<std::string, std::string` from `ov::AnyMap`.
Here we have an opposite thing.

Comment (ilya-lavrenov) [reply src/inference/src/ie_core.cpp:623]: I understand, that is why I'm asking about opposite function :)

Comment (ilyachur) [reply src/inference/src/ie_core.cpp]: Fixed

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**